src/averageAges.js

```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const allMen = people.filter((person) => person.sex === 'm');
+  const menFromThisCentury = century === undefined
+    ? allMen
+    : allMen.filter((man) => Math.ceil(man.died / 100) === century);
+
+  const menAge = menFromThisCentury.map((man) => man.died - man.born);
+  const sumOfMenAges = menAge.reduce((result, age) => result + age, 0);
+
+  return sumOfMenAges / menAge.length;
```

Comment:
Similar calculations are performed in the next function, take it to a separate function and reuse it.